src/pages/QuestionPage.jsx

```diff
@@ -227,7 +20,7 @@ const QuestionPage = () => {
 
   // State to track the status of each question (correct, incorrect, unanswered)
   const [questionStatus, setQuestionStatus] = useState(
-    Array(questionData.length).fill("unanswered"),
+    Array(currentCategory.questions.length).fill("unanswered"),
   );
 
```

Comment:
const [currentQuestionIndex, setCurrentQuestionIndex] = useState(0);
put this in context